core/modules/replica/python/schema_migration.py

```diff
@@ -0,0 +1,219 @@
+"""Module defining methods used in schema migration of qservReplica database.
+"""
+
+__all__ = ["make_migration_manager"]
+
+#--------------------------------
+#  Imports of standard modules --
+#--------------------------------
+import logging
+import os
+import re
+
+#-----------------------------
+# Imports for other modules --
+#-----------------------------
+from lsst.qserv.schema import SchemaMigMgr
+from lsst.db import utils
+
+#----------------------------------
+# Local non-exported definitions --
+#----------------------------------
+
+_log = logging.getLogger(__name__)
+
+#------------------------
+# Exported definitions --
+#------------------------
+
+
+class QservReplicaMigrationManager(SchemaMigMgr):
+    """Class implementing schema migration for qservReplica database.
+    """
+
+    # migration script has name "migrate-N-to-M.sql" or "migrate-N-to-M-something.sql"
+    # when applying scripts for any given version N the names are sorted first.
+    _mig_name_re = re.compile(r"migrate-(?P<from>\d+)-to-(?P<to>\d+)(-.*)?.sql")
+
+    def __init__(self, name, engine, scripts_dir):
+
+        SchemaMigMgr.__init__(self)
+
+        self.engine = engine
+
+        # scripts are located in replica/ sub-dir
+        scripts_dir = os.path.join(scripts_dir, name)
+
+        # find all migration scripts, add full script path to each
+        scripts = self.findScripts(scripts_dir, self._script_match)
+        self.scripts = [(v0, v1, fname, os.path.join(scripts_dir, fname))
+                        for v0, v1, fname in scripts]
+
+    @classmethod
+    def _script_match(cls, fname):
+        """Match script name against pattern.
+
+        Returns
+        -------
+        None if no match, pair of version numbers if there is match.
+        """
+        match = cls._mig_name_re.match(fname)
+        if match:
+            v_from = int(match.group('from'))
+            v_to = int(match.group('to'))
+            return (v_from, v_to)
+        return None
+
+    def current_version(self):
+        """Returns current schema version.
+
+        Returns
+        -------
+        Integer number
+        """
+
+        # Initial qservReplica implementation did not have version number stored at all,
+        # and we call this version 0. Since version=1 version number is stored in
+        # QMetadata table with key="version"
+        if not utils.tableExists(self.engine, "QMetadata"):
+            _log.debug("QMetadata missing: version=0")
```

Comment:
The comment explains it but the log message is too terse to be useful I think. I'd consider doing more explaining in the log message.